cli/src/cluster_query.rs

```diff
@@ -1817,8 +1813,40 @@ pub fn process_show_stakes(
 ) -> ProcessResult {
     use crate::stake::build_stake_state;
 
-    let progress_bar = new_spinner_progress_bar();
-    progress_bar.set_message("Fetching stake accounts...");
+    // fetching corresponding vote pubkey from identity pubkey
+    let vote_account_pubkeys = match vote_account_pubkeys {
+        Some(pubkeys) => {
+            let vote_account_progress_bar = new_spinner_progress_bar();
+            vote_account_progress_bar.set_message("Searching for matching vote accounts...");
+
+            let vote_accounts = rpc_client.get_vote_accounts()?;
+
+            let mut pubkeys: HashSet<String> =
+                pubkeys.iter().map(|pubkey| pubkey.to_string()).collect();
+
+            let vote_account_pubkeys: HashSet<String> = vote_accounts
+                .current
+                .iter()
+                .chain(vote_accounts.delinquent.iter())
+                .filter(|vote_acc| {
+                    pubkeys.remove(&vote_acc.node_pubkey) || pubkeys.remove(&vote_acc.vote_pubkey)
+                })
+                .map(|vote_acc| &vote_acc.vote_pubkey)
+                .cloned()
+                .collect();
```

Comment:
You changed a few parts of my suggestion here. Why?
`iter().cloned()` is less performant.

Reply:
sorry, I only saw your .remove() code, didn't pay attention to the rest of it. Updated it, thanks for your patience